src/messages_manager.ts

```diff
@@ -160,3 +159,27 @@ export class MessagesManager implements Disposable {
         return msg + "\n";
     }
 }
+
+class MessageBuffer {
```

Comment:
Defining a new class adds a layer of indirection. This seems a bit over-engineered. `handleFlush()` is where the logic lives, and `MessageBuffer` is essentially an indirect way of moving the logic out of `handleFlush()`. 

Is `MessageBuffer` a useful abstraction? `MessagesManager` itself is already an abstraction doing the dirty work. So moving some parts of it to a similar `MessageBuffer` is indirection without a clear semantic improvement.

Reply:
Yes. This should be unnecessary. I initially thought it was due to changes in Neovim's output, so I first tried modifying the content in the message. Later, I realized it was actually a change in VSCode, where the document in the output uses `\r\n` for line breaks on Windows.